examples/goldilocks_models/initial_guess.cc

```diff
@@ -0,0 +1,195 @@
+//
+// Created by jianshu on 3/25/20.
+//
+#include "examples/goldilocks_models/initial_guess.h"
+
+namespace dairlib::goldilocks_models {
+//edited by Jianshu to try a new way of setting initial guess
+
+MatrixXd get_theta_scale(const string directory, int iter){
+//    VectorXd initial_theta = readCSV(directory + to_string(0) + string("_theta_s.csv"));
+    VectorXd current_theta = readCSV(directory + to_string(iter) + string("_theta_s.csv"));
+    VectorXd theta_scale = 1*VectorXd::Ones(current_theta.rows());
+    MatrixXd diag_theta_scale = theta_scale.asDiagonal();
+    return diag_theta_scale;
+}
+
+MatrixXd get_gamma_scale(int gamma_length, double min_sl, double max_sl, double min_gi, double max_gi,
+        double min_tr, double max_tr){
+    VectorXd gamma_scale(gamma_length);
+    gamma_scale << 1, 1, 2;
+    MatrixXd diag_gamma_scale = gamma_scale.asDiagonal();
+    return diag_gamma_scale;
+}
+
+string set_initial_guess(const string directory, int iter, int sample, int total_sample_num,
+                         double min_sl, double max_sl, double min_gi, double max_gi, double min_tr,
+                         double max_tr) {
+    /* define some parameters used in interpolation
+* theta_range :decide the range of theta to use in interpolation
+* theta_sclae,gamma_scale :used to scale the theta and gamma in interpolation
+*/
+    double theta_range = 1; //Have not tuned yet!!
+    int gamma_dimension = 3;
+    MatrixXd theta_scale = get_theta_scale(directory, iter);
+    MatrixXd gamma_scale = get_gamma_scale(gamma_dimension, min_sl, max_sl, min_gi, max_gi, min_tr, max_tr);
+//    initialize variables used for setting initial guess
+    VectorXd initial_guess;
+    VectorXd weight_theta;
+    MatrixXd w_theta;
+    int past_iter;
+    int sample_num;
+//    get theta of current iteration and task of current sample
+    VectorXd current_theta = readCSV(directory + to_string(iter) + string("_theta_s.csv"));
+    MatrixXd current_ground_incline = readCSV(directory + to_string(iter) + string("_") + to_string(sample)
+                                              + string("_ground_incline.csv"));
+    MatrixXd current_stride_length = readCSV(directory + to_string(iter) + string("_") + to_string(sample)
+                                             + string("_stride_length.csv"));
+    MatrixXd current_turning_rate = readCSV(directory + to_string(iter) + string("_") + to_string(sample)
+                                             + string("_turning_rate.csv"));
+    VectorXd current_gamma(gamma_dimension);
+    current_gamma << current_ground_incline(0, 0), current_stride_length(0, 0), current_turning_rate(0,0);
+//    For iter = 1, theta_1 is the same with theta_0. So use the results of iter_0.
+    if(iter == 1)
```

Comment:
It will be nice to have some documentation in the beginning to describe algorithm, so that the readers don't need to decipher the codes themselves

E.g. If iter = 1, we look for the closest task sample and use the solution from that sample for initial guess.

Reply:
Question -- why do we use the nearest task sample instead of weighted sum of all samples?

Reply:
Good point.

Reply:
For iteration 1, using the interpolation of limited samples from iteration 0 might be worse than using the nearest sample.